test/unit/groupTest.js

```diff
@@ -93,6 +93,46 @@ test('changing dance moves for all cats updates only all cat dancers', async t =
   nativeAPI.reset();
 });
 
+test('changing dance moves for all to rand sets same dance for all dancers', async t => {
+  const nativeAPI = await helpers.createDanceAPI();
+  nativeAPI.play({
+    bpm: 120,
+  });
+
+  // Mock 4 cat animation poses
+  const moveCount = 4;
+  for(let i = 0; i < moveCount; i++) {
+    nativeAPI.setAnimationSpriteSheet("CAT", i, {}, () => {});
+    nativeAPI.world.MOVE_NAMES.push({
+      name: `move${i}`
+    })
+  }
+
+  nativeAPI.world.fullLengthMoveCount = moveCount;
+  nativeAPI.world.restMoveCount = 1;
+
+  let spriteGroup = [];
+  const groupCount = 5;
+  for(let i = 0; i < groupCount; i++){
```

Comment:
going to get some lint errors on your for loops i think once this is merged with your eslint changes